code/game/gamemodes/gang/vigilante.dm

```diff
@@ -0,0 +1,203 @@
+/obj/item/device/vigilante_tool
+	name = "Vigilant's Companion"
+	desc = "A reverse-engineered gang tool designed by Nanotrasen to encourage crew to resist gang activity."
+	icon_state = "gangtool-white"
+	item_state = "walkietalkie"
+	w_class = WEIGHT_CLASS_TINY
+	origin_tech = "programming=5;bluespace=2"
+	var/points = 0
+	var/list/tags = list()
+	var/vig_item_list
+	var/vig_category_list
+	var/static/gang = "Vigilante"
+	var/static/list/vigilante_items = list(
+		/datum/gang_item/function/implant,
+		/datum/gang_item/weapon/hatchet,
+		/datum/gang_item/weapon/pitchfork,
+		/datum/gang_item/weapon/improvised,
+		/datum/gang_item/weapon/ammo/buckshot_ammo,
+		/datum/gang_item/weapon/surgood,
+		/datum/gang_item/weapon/ammo/surplus_ammo,
+		/datum/gang_item/weapon/auto,
+		/datum/gang_item/weapon/ammo/auto_ammo,
+		/datum/gang_item/weapon/ammo/auto_ammo_AP,
+		/datum/gang_item/weapon/riot,
+		/datum/gang_item/weapon/ammo/buckshot_ammo,
+		/datum/gang_item/weapon/launcher,
+		/datum/gang_item/equipment/soap,
+		/datum/gang_item/equipment/sharpener,
+		/datum/gang_item/equipment/brutepatch,
+		/datum/gang_item/equipment/shield,
+		/datum/gang_item/equipment/bulletproof_armor,
+		/datum/gang_item/equipment/bulletproof_helmet,
+		/datum/gang_item/equipment/gangbreaker,
+		/datum/gang_item/equipment/seraph
+		)
+
+/obj/item/device/vigilante_tool/New(mob/user)
+	var/datum/action/innate/vigilante_tool/VT = new
+	VT.Grant(user, src)
+	vig_item_list = list()
+	vig_category_list = list()
+	for(var/V in vigilante_items)
+		var/datum/gang_item/G = new V()
+		vig_item_list[G.id] = G
+		var/list/Cat = vig_category_list[G.category]
+		if(Cat)
+			Cat += G
+		else
+			vig_category_list[G.category] = list(G)
+	addtimer(CALLBACK(src, .proc/income), 1800)
+
+/obj/item/device/vigilante_tool/proc/income()
+	if(!src)
+		return
+	var/list/all_territory
+	var/newpoints = 0
+	var/mob/living/carbon/human/H = src.loc
+	for(var/datum/gang/G in SSticker.mode.gangs)
+		all_territory += G.territory
+	for(var/area/A in tags)
+		if(A in all_territory)
+		else
```

Comment:
Ah yea that was a byproduct of me debugging, couldnt figure out the issue until I realized the tag's territory var and the actual area were slightly different because of how they were being defined - solved that so I can move it back to where I had it before.